hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenCreateHandler.java

```diff
@@ -242,18 +262,9 @@ private Token buildToken(
      * @param op token creation transaction body
      * @return given expiry metadata
      */
-    private ExpiryMeta getExpiryMeta(
-            final long consensusTime,
-            @NonNull final TokenCreateTransactionBody op,
-            @NonNull final HandleContext context) {
-        final var maxEntityLifetime =
-                context.configuration().getConfigData(EntitiesConfig.class).maxLifetime();
-        final var impliedExpiry = op.hasAutoRenewPeriod()
-                ? consensusTime + op.autoRenewPeriod().seconds()
-                : consensusTime + maxEntityLifetime;
-
+    private ExpiryMeta getExpiryMeta(@NonNull final TokenCreateTransactionBody op) {
         return new ExpiryMeta(
-                impliedExpiry,
+                op.expiry().seconds(),
                 op.hasAutoRenewPeriod() ? op.autoRenewPeriod().seconds() : NA,
                 // Shard and realm will be ignored if num is NA
                 op.autoRenewAccount());
```

Comment:
I see that in the old code, we are just passing the expiry and then we do:
`return then.isAfter(consensusNow) && then.isBefore(expiryGivenMaxLifetime);`
Why do we need this additional calculation?